reference/test/solver/ir_kernels.cpp

```diff
@@ -450,4 +450,43 @@ TYPED_TEST(Ir, RichardsonConjTransposedSolvesTriangularSystem)
 }
 
 
+TYPED_TEST(Ir, ApplyWithGivenHintIsEquivalentToRef)
+{
+    using Mtx = typename TestFixture::Mtx;
+    using value_type = typename TestFixture::value_type;
+    using input_hint = gko::solver::input_hint;
+    auto ref_solver =
+        gko::solver::Ir<value_type>::build()
+            .with_criteria(
+                gko::stop::Iteration::build().with_max_iters(1u).on(this->exec))
```

Comment:
Shouldn't 1 iteration already perform one iteration, meaning `x` should be modified slightly?

Reply:
I do not get it.
ref_solver and solver will perform 1 iteration update.
ref_solver takes x as initial guess like the original behavior.
but solver will add the hint (or initial_guess).
For example, ref_solver will take zero vector ref_x but solver will take nonzero vector x with zero guess.
Compare the result after one iteration

Reply:
You are correct, I thought you simply compared the `x` after running not a single iteration.
I somehow missed that you created another solver and checked the results against that.